packages/flame/lib/src/rendering/decorator.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:ui';
+
+import 'package:flame/src/rendering/paint_decorator.dart';
+
+/// [Decorator] is an abstract class that encapsulates a particular visual
+/// effect that should apply to drawing commands wrapped by this class.
+///
+/// The simplest way to apply a [Decorator] to a component is to override its
+/// `renderTree` method like this:
```

Comment:
Maybe there should be a mixin that does this override?

Reply:
Added `HasDecorator` class.